subsys/bluetooth/host/l2cap.c

```diff
@@ -376,9 +382,14 @@ static struct net_buf *l2cap_create_le_sig_pdu(struct net_buf *buf,
 					       u16_t len)
 {
 	struct bt_l2cap_sig_hdr *hdr;
+	struct net_buf_pool *pool = NULL;
+
+	if (code == BT_L2CAP_DISCONN_REQ) {
+		pool = &disc_pool;
+	}
 
 	/* Don't wait more than the minimum RTX timeout of 2 seconds */
-	buf = bt_l2cap_create_pdu_timeout(NULL, 0, K_SECONDS(2));
+	buf = bt_l2cap_create_pdu_timeout(pool, 0, K_SECONDS(2));
```

Comment:
Unrelated to this patch since it doesn't change it, but there's `#define L2CAP_DISC_TIMEOUT	K_SECONDS(2)` higher up in this file which should probably be used? Or is that for something else?

Reply:
That is something else, this applies to all the L2CAP signalling not a specific PDU.